pycbc/psd/variation.py

```diff
@@ -160,3 +165,220 @@ def find_trigger_value(psd_var, idx, start, sample_rate):
+                            high_freq):
+    """ Calculates time series of PSD variability
+
+    This function first splits the segment up into 512 second chunks. It
+    then calculates the PSD over this 512 second. The PSD is used to
+    to create a filter that is the composition of three filters:
+    1. Bandpass filter between f_low and f_high.
+    2. Weightining filter which gives the rough response of a CBC
+    template.
+    3. Whitening filter.
+    Next it makes the convolution of this filter with the stretch of data.
+    This new time series is given to the "mean_square" function, which
+    computes the mean square of the timeseries within an 8 seconds window,
+    once per second.
+    The result, which is the variance of the S/N in that stride for the
+    Parseval theorem, is then stored in a timeseries.
+
+    Parameters
+    ----------
+    strain : TimeSeries
+        Input strain time series to estimate PSDs
+    segment : {float, 8}
+        Duration of the segments for the mean square estimation in seconds.
+    short_segment : {float, 0.25}
+        Duration of the short segments for the outliers removal.
+    psd_long_segment : {float, 512}
+        Duration of the long segments for PSD estimation in seconds.
+    psd_duration : {float, 16}
+        Duration of the segments for PSD estimation in seconds.
+    psd_stride : {float, 8}
+        Separation between PSD estimation segments in seconds.
+    psd_avg_method : {string, 'median'}
+        Method for averaging PSD estimation segments.
+    low_freq : {float, 20}
+        Minimum frequency to consider the comparison between PSDs.
+    high_freq : {float, 480}
+        Maximum frequency to consider the comparison between PSDs.
+
+    Returns
+    -------
+    psd_var : TimeSeries
+        Time series of the variability in the PSD estimation
+    """
+    # Calculate strain precision
+    if strain.precision == 'single':
+        fs_dtype = numpy.float32
+    elif strain.precision == 'double':
+        fs_dtype = numpy.float64
+
+    # Convert start and end times immediately to floats
+    start_time = numpy.float(strain.start_time)
+    end_time = numpy.float(strain.end_time)
+
+    # Resample the data
+    strain = resample_to_delta_t(strain, 1.0/2048)
+    srate = int(strain.sample_rate)
+
+    # Fix the step for the PSD estimation and the time to remove at the
+    # edge of the time series.
+    step = 1.0
+    strain_crop = 8.0
+
+    # Find the times of the long segments
+    times_long = numpy.arange(start_time, end_time,
+                              psd_long_segment-2*strain_crop
+                              - segment + step)
+
+    # Set up the empty time series for the PSD variation estimate
+    psd_var = TimeSeries(zeros(int(numpy.floor((end_time - start_time -
+                                                2*strain_crop - segment +
+                                                1) / step))),
```

Comment:
this formatting is hard to read. split up into 2 lines .. 
eg first line calculate `ts_duration = end_time - start_time - 2*strain_crop - segment + 1`, second line divide this by `step` and create the TimeSeries